SimpleSnmp.cs

```diff
@@ -966,10 +1326,10 @@ public bool Valid
 					val = GetBulk(new string[] { lastOid.ToString() });
 				}
 				// check that we have a result
-				if (val == null || val.Count == 0)
+				if (val == null)
 				{
 					// error of some sort happened. abort...
-					break;
+					return null;
```

Comment:
do not change this, it will may break someones' work flow.
I prefer to continue with the Walk when error happend, see https://github.com/rqx110/SnmpSharpNet/commit/7127fc2c57f15bb64c459f61d12c5f7fc2e078fc